### PR TITLE
Player

### PR DESCRIPTION
**Pull request by chashafe**

### CODE REVIEW

File: Player.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        /*-----*/
+        public bool GetIsPlayerInHand() => isPlayerInHand; 
+        internal void ChangeIsPlayerInHand() => isPlayerInHand = true;
+
+        public bool GetIsPlayerInGame() => isPlayerInGame;
+        public void ChangeIsPlayerInGame() => isPlayerInGame = true;
+
+        public void AddCardToPlayerCards(Card SingleCard) => Cards.Insert(0, SingleCard);
+        public void RemoveCardFromePlayerCards(Card SingleCard) => new PlayerDeck().RemoveTopCard();
+        public bool CheckIfLoose(List<Card> Cards) => (Cards.Count < 1) ? true : false;
```
Comment (ntiany) [inline Player.cs]: you don't need ? true : false the statement (Cards.Count < 1) itself will return true or false

File: Player.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+
+
+namespace Card_Game
+{
+    public class Player 
+    {
+        bool isPlayerInHand = false;  // na true, jeśli prowadzi grę
+        bool isPlayerInGame = false; // na true, jeśli jeszcze w grze
+        List<Card> Cards {get; set;}
```
Comment (AgnieszkaChS) [inline Player.cs]: How about Player having a PlayerDeck instead of Cards? Then we could use deck's methods in it.

File: Player.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }
+        /*-----*/
+        public bool GetIsPlayerInHand() => isPlayerInHand; 
+        internal void ChangeIsPlayerInHand() => isPlayerInHand = true;
+
+        public bool GetIsPlayerInGame() => isPlayerInGame;
+        public void ChangeIsPlayerInGame() => isPlayerInGame = true;
+
+        public void AddCardToPlayerCards(Card SingleCard) => Cards.Insert(0, SingleCard);
+        public void RemoveCardFromePlayerCards(Card SingleCard) => new PlayerDeck().RemoveTopCard();
```
Comment (ntiany) [inline Player.cs]: Should be "From" instead of "Frome"

File: Player.cs (hunk 2 above)
Comment (chashafe) [reply Player.cs]: thx

File: Player.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+
+
+namespace Card_Game
+{
+    public class Player 
+    {
+        bool isPlayerInHand = false;  // na true, jeśli prowadzi grę
```
Comment (ntiany) [inline Player.cs]: it could be a property public get; private set;

File: Player.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        bool isPlayerInGame = false; // na true, jeśli jeszcze w grze
```
Comment (ntiany) [inline Player.cs]: it could be a property public get; private set; (then you won't need methods to get this value)

)

File: Player.cs (hunk 2 above)
Comment (ntiany) [inline Player.cs]: it should be a field if it's fully private member

File: Player.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+
+
+namespace Card_Game
+{
+    public class Player 
+    {
```
Comment (AgnieszkaChS) [inline Player.cs]: From Table class perspective it would be cool to have IsRoundWinner property in Player Class

File: Player.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        bool isPlayerInGame = false; // na true, jeśli jeszcze w grze
+        List<Card> Cards {get; set;}
+        /*----*/
+        public Player()
+        {
+            Cards = new List<Card>();
+        }
+        /*-----*/
+        public bool GetIsPlayerInHand() => isPlayerInHand; 
+        internal void ChangeIsPlayerInHand() => isPlayerInHand = true;
```
Comment (ntiany) [inline Player.cs]: why internal?

Comment (ntiany) [reply Player.cs]: we will need a method to change isPlayerInHand to false

File: Player.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+        public bool GetIsPlayerInGame() => isPlayerInGame;
+        public void ChangeIsPlayerInGame() => isPlayerInGame = true;
+
+        public void AddCardToPlayerCards(Card SingleCard) => Cards.Insert(0, SingleCard);
```
Comment (AgnieszkaChS) [inline Player.cs]: AddCard, RemoveCard and CheckIfLoose methods can base on Deck's methods (e.g. IsEmpty method) if Player has a PlayerDeck

File: Player.cs (hunk 6 above)
Comment (AgnieszkaChS) [reply Player.cs]: And maybe Name prop for the display purpose?

File: Player.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            if (SingleCard is null)
+            {
+                throw new System.ArgumentNullException(nameof(SingleCard));
+            }
+            localDeck.AddCardToDeckBottom(SingleCard);
+        }
+
+        public void AddCardToLocalDeck(Card SingleCard) => localDeck.AddCardToDeckBottom(SingleCard);
+        public void RemoveCardFromePlayerCards(Card SingleCard) => localDeck.RemoveTopCard();
```
Comment (AgnieszkaChS) [inline Player.cs]: No need for SingleCard parameter, as RemoveTopCard() doesn't need one

File: Player.cs [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        public void AddCardToPlayerCards(Card SingleCard)
+        {
+            if (SingleCard is null)
+            {
+                throw new System.ArgumentNullException(nameof(SingleCard));
+            }
+            localDeck.AddCardToDeckBottom(SingleCard);
+        }
+
+        public void AddCardToLocalDeck(Card SingleCard) => localDeck.AddCardToDeckBottom(SingleCard);
```
Comment (AgnieszkaChS) [inline Player.cs]: Isn't AddCardToPlayerCards(Card SingleCard) the same as AddCardToLocalDeck(Card SingleCard)?

Comment (chashafe) [reply Player.cs]: of course :)

File: Player.cs (hunk 9 above)
Comment (chashafe) [reply Player.cs]: ok

**Review by starczi — APPROVED** [review]

**Review by ntiany — APPROVED** [review]

**Pull request merged**